python/ctsm/site_and_regional/single_point_case.py

```diff
@@ -0,0 +1,625 @@
+        weight or percentage of each pft.
+    num_pft : list
+        total number of pfts for surface dataset (if crop 78 pft, else 16 pft)
+    zero_nonveg_landunits : bool
+        flag for setting all non-vegetation landunits to zero
+    uni_snow : bool
+        flag for creating datasets using uniform snowpack
+    saturation_excess : bool
+        flag for making dataset using saturation excess
+
+    Methods
+    -------
+    create_tag
+        create a tag for single point which is the site name
+        or the "lon-lat" format if the site name does not exist.
+
+    create_domain_at_point
+        Create domain file at a single point.
+
+    create_landuse_at_point:
+        Create landuse file at a single point.
+
+    create_surfdata_at_point:
+        Create surface dataset at a single point.
+
+    create_datmdomain_at_point:
+        Create DATM domain file at a single point.
+
+    extract_datm_at:
+        Extract DATM for one file at a single point.
+
+    create_datm_at_point:
+        Extract all DATM data at a single point.
+    """
+
+    # pylint: disable=too-many-instance-attributes
+    # the ones we have are useful
+
+    def __init__(
+            self,
+            plat,
+            plon,
+            site_name,
+            create_domain,
+            create_surfdata,
+            create_landuse,
+            create_datm,
+            create_user_mods,
+            dom_pft,
+            pct_pft,
+            num_pft,
+            include_nonveg,
+            uni_snow,
+            cap_saturation,
+            out_dir,
+    ):
+        super().__init__(create_domain, create_surfdata, create_landuse, create_datm,
+                         create_user_mods)
+        self.plat = plat
+        self.plon = plon
+        self.site_name = site_name
+        self.dom_pft = dom_pft
+        self.pct_pft = pct_pft
+        self.num_pft = num_pft
+        self.include_nonveg = include_nonveg
+        self.uni_snow = uni_snow
+        self.cap_saturation = cap_saturation
+        self.out_dir = out_dir
+
+        self.create_tag()
+        self.check_dom_pft ()
+        self.check_nonveg ()
+        self.check_pct_pft ()
+
+    def create_tag(self):
+        """
+        Create a tag for single point which is the site name
+        or the "lon-lat" format if the site name does not exist.
+        """
+        if self.site_name:
+            self.tag = self.site_name
+        else:
+            self.tag = "{}_{}".format(str(self.plon), str(self.plat))
+
+    def check_dom_pft (self):
+        """
+        A function to sanity check values in dom_pft:
+
+        - Compare dom_pft (values if more than one) with num_pft:
+          i.e. If dom_pft is 18 without crop it fails.
+
+        - Check for mixed land-units:
+          If we have more than one dom_pft, they should be in the
+          same range.
+          e.g. If users specified multiple dom_pft, they should be
+          either in :
+            - 1-14 range
```

Comment:
Thanks for your comment Erik. 
There are essentially two components here:
1) min `dom_pft` should be zero and not one. 
_I have implemented this now, so now the user can use --dompft 0 too._ 
I am not sure if it makes sense or not, but either way we added it to this in case the user wants bare-ground.


2) Updating the docstrings to use the constants instead of values, from:

```
either in :
            - 1-14 range
            or
            - 15-78 range
```
to 

```
either in 
        - 0 - NAT_PFT-1 range (i.e. 0-14)
         or
        - NAT_PFT - MAX_PFT range (i.e. 15-78)
```

I intentionally kept the ranges there too, so the user won't be confused. For example 0-NAT_PFT-1 is a bit confusing to read, so I added (i.e. 0-14).
 
I updated in some other docstrings too:
```
-                dom_pft values are both in range of 1-14 and 15-78.
+                dom_pft values are both in range of 0- NAT_PFT-1 (i.e. 0-14) and 15-78 (i.e. NAT_PFT - MAX_PFT).

```

I think both points are addressed and implemented. Thanks for your comment.

Reply:
I updated it further to look like this:
```
- 0 - NAT_PFT-1 range
or
- NAT_PFT - MAX_PFT 
``` and did some further updates on clearing up the docstrings and error messages.

Reply:
This looks great, I like the way this looks now.